sim/lib/core/include/core/systemContext.h

```diff
@@ -4,14 +4,19 @@
 #include <forward_list>
 #include <functional>
 #include "ics/componentStore.h"
-#include "ics/indexStore.h"
 #include "graphicsContext.h"
 
-typedef std::function<void(GraphicsContext&, ics::ComponentStore&, ics::IndexStore&)> SystemFn;
+template<class IndexStore>
```

Comment:
No sure if templating is the good idea. Can result in undesirable code like `SystemFn<int>` for example. Maybe a `IndexStore` interface with different `IndexStore` implementations could be better.

Reply:
Resolved in b41a9ab.